src/Caching/Config/FileHashComputer.php

```diff
@@ -17,8 +17,7 @@ public function compute(string $filePath): string
     {
         $this->ensureIsPhp($filePath);
 
-        $parametersHash = SimpleParameterProvider::hash();
-        return sha1($filePath . $parametersHash . VersionResolver::PACKAGE_VERSION);
+        return sha1($filePath . VersionResolver::PACKAGE_VERSION);
```

Comment:
`$parametersHash` is collected when rules added/removed as well, so that is needed to rebuild the cache, so in case of command arg, the ARGV may probably needed for that detection.

Reply:
as far as I understood the file-cache, it stores 1:1 the content of the files as a hash.

this hash is independent from the rector config/rules. why do we need to invalidate it ?

Reply:
for example, the 1st run, we have the config:

```php
$rectorConfig->rules([A::class]);
```

then, on 2nd run, we have updated config:

```diff
-$rectorConfig->rules([A::class]);
+$rectorConfig->rules([A::class, B::class]);
```

so the file cache need to be invalidated, so the run is applied instead of ignored in case of `B::class` rule can make the change.

Reply:
sure, but this happens because of hash of the rector.php changed (also after the change of this PR)

Reply:
I am thinking of different case, for example, we run the multiple steps:

1. run into `rector process app/Model` : no change
2. run into `rector process app/Services`: has change, that can be used by app/Model
3. run into `rector process app/Model`: it should make a change because of `app/Services` was changed, but because of this change, it cached, and not applied

so, the old way seems have valid use case.

Reply:
when the command `rector process app/Services` leads to a change of `app/Model` rector should already update the cache for said file?

Reply:
that's not the case, rector can rely on other file, eg on other method by other paths, eg the following rules:

- ParamTypeByMethodCallTypeRector
- AddMethodCallBasedStrictParamTypeRector

on first run on `app/Model`, it not changed, then cached.
on second run on `app/Services`, it changed other paths, then cached.
back flip to run `app/Model`, it should allow change as changed typed in previous `app/Services`

Reply:
no e2e test case for that yet, but that can happen, as for the example, param is rely on external type. 

I think if you want to use cache, use full project  (`vendor/bin/rector`) in single ci process is better as whole project checked instead of make it multiple CIs.

Reply:
> I think if you want to use cache, use full project (`vendor/bin/rector`) in single ci process is better as whole project checked instead of make it multiple CIs.

we need separate rector invocations as we have non-unique class names and adding PSR4 is not feasible.

see e.g. our composer script
```
        "rector": [
            "vendor/bin/rector process application/B2B/ -a application/B2B/phpstan-bootstrap.php --ansi",
            "vendor/bin/rector process application/B2Brocket/ -a application/B2Brocket/phpstan-bootstrap.php --ansi",
            "vendor/bin/rector process application/cli/ -a application/cli/phpstan-bootstrap.php --ansi",
            "vendor/bin/rector process application/Rocket/ -a application/Rocket/phpstan-bootstrap.php --ansi",
            "vendor/bin/rector process application/api/ config/ library/ public/ scripts/ ecs.php deploy.php rector.php --ansi"
        ]
```

it runs within a single github action. its not distributed over multiple CI jobs.

Reply:
since we have a workarround which works for us, I will close this PR here.

I think my time is better invested somewhere else. still I think the cases you mentioned should have a test-case and having a way to run several different rector commands without the cache beeing cleared would be very helpful.

thanks for the feedback abdul!